src/main/java/com/kodilla/ecommercee/controller/ProductController.java

```diff
@@ -1,42 +1,54 @@
 package com.kodilla.ecommercee.controller;
 
 import com.kodilla.ecommercee.domain.dto.ProductDto;
-import org.springframework.web.bind.annotation.*;
+import com.kodilla.ecommercee.mapper.ProductMapper;
+import com.kodilla.ecommercee.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
 
-import java.math.BigDecimal;
-import java.util.ArrayList;
 import java.util.List;
 
 @RestController
 @RequestMapping("/products")
 public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
 
-    @GetMapping(value = "getProducts")
-    public List<ProductDto> getProducts() {
-        List<ProductDto> products = new ArrayList<>();
-        products.add(new ProductDto(4L, "test name 3", "test description 3", new BigDecimal("19.32"), 43L));
-        products.add(new ProductDto(8L, "test name 4", "test description 4",  new BigDecimal("424.12"), 23L));
+    @Autowired
+    public ProductController(ProductService productService, ProductMapper productMapper) {
+        this.productService = productService;
+        this.productMapper = productMapper;
+    }
 
-        return products;
+    @GetMapping
+    public List<ProductDto> getProducts() {
+        return productMapper.mapToProductDtoList(productService.getProductList());
     }
 
-    @GetMapping(value = "getProduct/{id}")
+    @GetMapping(value = "{id}")
     public ProductDto getProduct(@PathVariable("id") Long productId) {
-        return new ProductDto(1L, "test name", "test description", new BigDecimal("9.99"), 2L);
+        return productMapper.mapToProductDto(productService.getProductById(productId));
     }
 
-    @PostMapping(value = "addProduct")
-    public void addProduct(@RequestBody ProductDto productDto) {
-
+    @PostMapping
+    public void addProduct(@RequestBody ProductDto productDto)  {
+        productService.addProduct(productMapper.mapToProduct(productDto));
     }
```

Comment:
Mam problem z metodą addProduct, próbowałem ją przetestować w Postmanie ale otrzymuję efekt jak na obrazku. A wydaję mi się że zapytanie jest ok.

![test](https://user-images.githubusercontent.com/43109258/53114212-988d7880-3543-11e9-8bd8-12b60d80cff0.jpg)

Reply:
W zapytaniu nie powinno być /addProduct

Reply:
Próbowałem też bez tego, nic nie zmienia. Może w body coś jest nie tak, no nie wiem.

![test3](https://user-images.githubusercontent.com/43109258/53117321-ef4a8080-354a-11e9-8126-c9aa146bdd46.jpg)

Reply:
Szczerze mówiąc teraz też nie wiem czemu nie działa. U mnie jest ok. Zobaczymy jak będzie u innych.
![post](https://user-images.githubusercontent.com/43876594/53118370-3c2f5680-354d-11e9-9816-da5e5bf27282.png)

Reply:
Dziwne, przed poprawkami Twoje endpointy działały u mnie a teraz też mam taki błąd jak na screenshotach

Reply:
A spróbowaliście tej metody bez id w JSONie? Ponieważ u mnie zdaje się działać
![przechwytywanie2](https://user-images.githubusercontent.com/41995059/53127164-21b3a800-3562-11e9-998b-4e5f9f3ed541.PNG)

![przechwytywanie3](https://user-images.githubusercontent.com/41995059/53127195-32641e00-3562-11e9-9a06-a58a520f4dfd.PNG)

Reply:
Bez id ma więcej sensu, bo jest ono i tak generowane automatycznie. U mnie obie metody działają

Reply:
Kilka razy też już mi nie działały pewne zapytania przez bałagan w bazie danych, a jak stworzyłem nową, to wszystko zaczęło działać prawidłowo

Reply:
Jak zrobiłem drop table to wszystko mi działało